src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -401,6 +402,18 @@ public void Dispose()
             {
                 var exceptionAggregator = new ExceptionAggregator(_logger, "Failure to dispose landlord");
 
+                // we don't want to wake up database during dispose.
+                var handles = new List<WaitHandle>();
+                foreach (var timer in _wakeupTimers.Values)
+                {
+                    var handle = new ManualResetEvent(false);
+                    timer.Dispose(handle);
+                    handles.Add(handle);
+                }
+
+                if (handles.Count > 0)
+                    WaitHandle.WaitAll(handles.ToArray());
```

Comment:
Is it possible that we have more than 64 items here?

Reply:
Theoretically, if we have more than 64 backup tasks, then yes.

Reply:
IIRC, `WaitAll` can accept a maximum number, need to handle this here.